src/components/ui/UITypography/UITypography.tsx

```diff
@@ -1,9 +1,24 @@
+/* eslint-disable react/jsx-props-no-spreading */
 import { ThemeProvider, Typography, createTheme } from '@mui/material';
+import { TypographyProps } from '@mui/material/Typography';
 
-import { UITypographyProps } from './UITypographyTypes';
+declare module '@mui/material/Typography' {
```

Comment:
this code isn't must be in the component, create Types.d.ts for this reason in UITypography folder